nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/aggregate/DateHistogramAggregate.java

```diff
@@ -150,47 +154,55 @@ public void parseEsBuckets(Collection<? extends MultiBucketsAggregation.Bucket>
     }
 
     private DateTime addInterval(DateTime from) {
-        return new DateTime(from.getMillis() + getIntervalInMillis());
+        String interval = getInterval();
+        switch (interval.toLowerCase()) {
+            case "second":
+                return from.plusSeconds(1);
+            case "minute":
+                return from.plusMinutes(1);
+            case "hour":
+                return from.plusHours(1);
+            case "day":
+                return from.plusDays(1);
+            case "week":
+                return from.plusDays(1);
+            case "month":
+                return from.plusMonths(1);
+            case "quarter":
+                return from.plusMonths(3);
+            case "year":
+                return from.plusYears(1);
+        }
+        if (interval.endsWith("y")) {
```

Comment:
pattern matching would make this nicer since we could return better error messages and would prevent NumberFormatException with mistakes like "days" which will fall through the switch but it works.